build/webTestReporter.js

```diff
@@ -73,24 +78,131 @@ exports.startReportServer = async function () {
 exports.dumpTestSummary = () => {
     try {
         const summary = JSON.parse(fs.readFileSync(webTestSummaryJsonFile).toString());
-        const eventEmitter = new EventEmitter();
-        const reportWriter = new mocha.reporters.Spec(eventEmitter);
+        const runner = new EventEmitter();
+        runner.stats = {};
+        const reportWriter = new mocha.reporters.Spec(runner, { color: true });
         reportWriter.failures = [];
+        const cells = [];
+        let indent = 0;
+        let executionCount = 0;
+        const skippedTests = [];
+        mocha.reporters.Base.useColors = true;
+        colors.enable();
         summary.forEach((output) => {
+            output = JSON.parse(JSON.stringify(output));
             // mocha expects test objects to have a method `slow, fullTitle, titlePath`.
-            ['slow', 'fullTitle', 'titlePath'].forEach((fnName) => {
+            ['slow', 'fullTitle', 'titlePath', 'isPending', 'currentRetry'].forEach((fnName) => {
                 const value = output[fnName];
                 output[fnName] = () => value;
             });
+            // Tests have a parent with a title, used by xunit.
+            const currentParent = output.parent || { fullTitle: '' };
+            output.parent = {
+                fullTitle: () => ('fullTitle' in currentParent ? currentParent.fullTitle : '') || ''
+            };
             if ('stats' in output) {
-                reportWriter.stats = output.stats;
+                reportWriter.stats = { ...output.stats };
+                Object.assign(runner.stats, output.stats);
             }
             if (output.event === 'fail') {
                 reportWriter.failures.push(output);
             }
-            eventEmitter.emit(output.event, Object.assign({}, output));
+            runner.emit(output.event, output, output.err);
+
+            switch (output.event) {
```

Comment:
Generate an ipynb file with:
* failed tests
* logs for each of the fail tests
* screenshots of the failed tests